### ISSUE TITLE
feature request: rename entry

### ISSUE THREAD

**Issue by as3ii** [doy/rbw#35]

**Comment by doy** [comment]
can you say more about what you'd like here?

**Comment by as3ii** [comment]
I think that could be cool to have the possibility to edit the entries, such as rename an entry or edit the username.
![2021-02-19_18-31](https://user-images.githubusercontent.com/29260716/108539808-5fc23e80-72d8-11eb-9b70-d6d348e63e17.png)
^ this is an example of the _item editor_ available inside the web client, the fields "Name" and "Username" are the ones I was talking about

**Comment by rjc** [comment]
Hello,

Commenting here rather than opening a new issue as, essentially, the OP is describing the problem I have.

I have created a new entry with:

    rbw add ...

but forgot to add a username - there's no way to add it post factum. Similar issue would be if I had done a typo anywhere but the password.

Also, given that `rbw` supports TOTP, it would be great if one could add/edit the secret, too. Side note, any chance to rename `code` to `totp` or at least provide the latter as an alias? `code` seems to generic a term and Bitwarden doesn't support anything else but TOTP.

To sum it up, it would be great if most (all?) fields were editable as is the case with, i.e. `lastpass-cli`.

Whilst the title of this issue reads _feature request_, it seems like this (basic?) functionality is simply missing, at least to me. I guess what I'm trying to say is that, like with other fully-featured command line software, i.e. `mutt`, `lastpass-cli`, etc., I don't want to use WebUI/GUI at all! :^)

Regards,

Raf

**Comment by rjc** [comment]
Hello,

Are there any prospects of getting this functionality in future releases?

Cheers,

Raf

**Comment by dezeroku** [comment]
I bumped into this when trying to change username/email address for one of the entries. I'd be happy to drop a PR for this.

The biggest problem here seems to be parsing the input data. I'm not a big fan of forcing users to edit JSONs, so maybe we could use YAMLs? So on the call to `rbw edit` we would open the editor with the existing fields prepopulated, e.g.:
```
password: some-password
username: some-username
uri:
  - https://example.com
```
user then would edit the specific fields and on the rbw's side we would iterate over the keys returned and apply changes only for them.

So with this implementation you could do something like this to modify a single field in the entry:
```
echo "password: new-password" | rbw edit some-entry
```
and something like this to remove a single field:
```
echo "some_entry: null" | rbw edit some-entry
```

WDYT?

**Comment by cjshearer** [comment]
The `get` command already supports a `--field` flag, so why not a matching flag for `edit`?

**Comment by fullstopslash** [comment]
Yeah, this feature is sorely missing. The fact i can easily retrieve, but not edit fields Is pretty confusing. Love this tool though! thank you for the work!
